### DISCUSSION TITLE
Is file extension language detection the only option?

### DISCUSSION THREAD

**Discussion by jcubic** [github-linguist/linguist#5694] [Q&A] [answered]
I've just renamed my PEG parser file from pegjs to peg, because I've switched to PEG.js fork Peggy and it looks odd to have the file named pegjs.

Is this library detect langauges only based on extension? If so this is pretty trivial you only need to have a mapping extension => langauge. If no then is there a way to make it detect the PEG.js grammar if it not have pegjs extension?

https://github.com/jcubic/gaiman/blob/master/src/grammar.peg

GitHub say that it's 99.5% written in JavaScript and ignore .peg file.

**Comment by lildude** [comment, marked as answer]
> Is this library detect langauges only based on extension? 

No. It also looks at things like filename and modeline comments too.

> If no then is there a way to make it detect the PEG.js grammar if it not have pegjs extension?

Yes, the [documentation](https://github.com/github/linguist/tree/master/docs) details how you can implement various different overrides to set the language.
